project_event/models/project_project.py

```diff
@@ -154,6 +150,8 @@ def confirm_accept_reservation(self):
                 activity.send_message('requested')
             activity.open_resources_reservation()
             activity.write({'task_state': 'accepted'})
+        if self.state in ['draft', 'option', 'postponed', 'canceled']:
+            self.send_message('accepted')
```

Comment:
@eilst  send_message() have only statement  self.env['mail.message'].create(self.get_message(action)) . Why you do not think about generic approach for all project.task which have the similar  syntax and format of referiing self.env[]